cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceContext.java

```diff
@@ -193,4 +191,9 @@
    * @param resources Resources that each reducer should use.
    */
   void setReducerResources(Resources resources);
+
+  /**
+   * Return true if the execution was successful, false otherwise.
+   */
+  boolean isSuccessful();
```

Comment:
This doesn't tell why it is not succeeded (program error vs explicit stop) and it doesn't have the exception if fails because of program error. Probably better to have something like `getCompletionState`, which return an object that can carry the state (running, completed, failed, killed) and an optional failure cause.